wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/urlfactory/URLFactoryBuilder.java

```diff
@@ -86,10 +84,20 @@ public URLFactoryBuilder(
       FileContentProvider fileContentProvider,
       Map<String, String> overrideProperties)
       throws BadRequestException {
+    String devfileYamlContent;
     for (DevfileLocation location : remoteFactoryUrl.devfileFileLocations()) {
-      String devfileYamlContent = urlFetcher.fetchSafely(location.location());
-      if (isNullOrEmpty(devfileYamlContent)) {
+      try {
+        devfileYamlContent = fileContentProvider.fetchContent(location.location());
+      } catch (IOException ex) {
+        // try next location
```

Comment:
can we at least log.debug this exception ?

Reply:
Added